beta/nncf/tensorflow/initialization.py

```diff
@@ -0,0 +1,59 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Union
+
+import tensorflow as tf
+
+from nncf.common.initialization import NNCFDataLoader
+from nncf.config.structure import BNAdaptationInitArgs
+from nncf.config import NNCFConfig
+
+
+class TFInitializingDataLoader(NNCFDataLoader):
+    """
+    This class wraps the tf.data.Dataset class.
+    This is required for proper initialization of certain compression algorithms.
+    """
+    def __init__(self, data_loader: tf.data.Dataset, batch_size: int):
+        self._data_loader = data_loader
+        self._batch_size = batch_size
+
+    @property
+    def batch_size(self) -> int:
+        return self._batch_size
+
+    def __iter__(self):
+        return iter(self._data_loader)
+
+
+def register_default_init_args(nncf_config: NNCFConfig,
+                               data_loader: Union[tf.data.Dataset, TFInitializingDataLoader],

```

Comment:
```suggestion
                               data_loader: tf.data.Dataset,
```
PR 744 hasn't been merged yet, so let handle it there.